### ISSUE TITLE
[video_player] is not working on IOS 16+ for HLS (.m3u8) videos

### ISSUE THREAD

**Issue by SamerOrfali22** [flutter/flutter#116021]
HLS videos (.m3u8) is not working on IOS devices which has OS 16.0 and above same video are working on Android devices and also on IOS under OS 16. 
flutter_version : 3.3.9 
video_player : 2.4.8 
test device : iPhone 13 pro, OS 16.1.1

to test, I've created simple flutter project only contain video_player and I used VideoPlayerController.network();
the console error: Unhandled Exception: PlatformException(VideoError, Failed to load video: The operation couldn’t be completed. (CoreMediaErrorDomain error -12646.), null, null)

also I find out that only encrypted HLS videos are not working.

**Comment by huycozy** [comment]
Hi @SamerOrfali22 
Can you share the video url so that we may verify it?

There is a related issue at #32029 (please take a look at https://github.com/flutter/flutter/issues/32029#issuecomment-1325570209). 
Also, a similar issue #115422 has been filled recently on Android platform. But the behavior is different on some iOS versions/devices:
- iPhone 14 Pro Max (iOS 16.0): can hear the sound but visual is not displayed.
- iPhone 7, iOS 15.7: same as your report, the video is playing well (demo at https://github.com/flutter/flutter/issues/115422#issuecomment-1319546706)

**Comment by SamerOrfali22** [comment]
@huycozy this is my link https://learner.ta3leem.iq/videos/playlist/course-v1:Ta3leem+Bio_12G_ramla+2022/93a47637-4631-4a2e-9fd6-7d4334719e0d.m3u8
and it's worked on IOS 16.0 but not worked no IOS 16.1

**Comment by hellohuanlin** [comment]
As explained in the [design doc](https://docs.google.com/document/d/1zelzxGf6kUEvtF6hbFRExs5UOz8ODtUCa7ogiouxbpo/edit#heading=h.5fof4ki35ptt): 

> It is pretty obvious that Apple's intention is to only allow its own video player (that is, AVPlayerLayer) to access those pixels of encrypted streams from iOS 16 onwards. The short term fix relies on a design flaw in AVFoundation, which could be fixed by Apple in future OS updates. 
> 
> For the long term plan, to follow Apple's spirit, the video_player plugin should directly expose a platform view (with AVPlayerLayer attached), instead of accessing the pixel buffers. 


(though I have not tried out and verified the latest iOS version yet).

**Comment by SamerOrfali22** [comment]
@hellohuanlin your solution works for IOS 16.0 but not working on 16.1 and above

**Comment by hellohuanlin** [comment]
@SamerOrfali22 I looked into this. My previous fix resolves still works on iOS 16.1.

I tried out your video link on SwiftUI and it fails to load: 

```swift
struct ContentView: View {
    var body: some View {
      VideoPlayer(player: AVPlayer(url: URL(string: "https://learner.ta3leem.iq/videos/playlist/course-v1:Ta3leem+Bio_12G_ramla+2022/93a47637-4631-4a2e-9fd6-7d4334719e0d.m3u8")!))
        .frame(height: 500)
    }
}
```
So it's not an issue on video_player plugin, but instead Apple's issue. I am gonna close this issue. You may want to file a radar with Apple.
